packages/gatsby/src/commands/build.ts

```diff
@@ -698,8 +698,8 @@ module.exports = async function build(
   }
 
   if (adapterManager) {
-    await adapterManager.adapt()
     await adapterManager.storeCache()
+    await adapterManager.adapt()
```

Comment:
`storeCache` first so we cache `public` directory before any potential file moves, so those assets are in places that gatsby expects on future builds